go/libraries/doltcore/sqle/tables.go

```diff
@@ -182,6 +182,65 @@ func (t *DoltTable) Delete(ctx *sql.Context, sqlRow sql.Row) error {
 	return nil
 }
 
+func (t *DoltTable) Update(ctx *sql.Context, oldRow sql.Row, newRow sql.Row) error {
+	dOldRow, err := SqlRowToDoltRow(t.table.Format(), oldRow, t.sch)
+	if err != nil {
+		return err
+	}
+	dNewRow, err := SqlRowToDoltRow(t.table.Format(), newRow, t.sch)
+	if err != nil {
+		return err
+	}
+	deleteFirst := false
+
+	// If the PK is changed then we have to delete the old row first
+	// This is assuming that the new PK is not taken
+	dOldKey := dOldRow.NomsMapKey(t.sch)
+	dOldKeyVal, err := dOldKey.Value(ctx)
+	if err != nil {
+		return err
+	}
+	dNewKey := dNewRow.NomsMapKey(t.sch)
+	dNewKeyVal, err := dNewKey.Value(ctx)
+	if err != nil {
+		return err
+	}
+	if !dOldKeyVal.Equals(dNewKeyVal) {
+		_, rowExists, err := t.table.GetRow(ctx, dNewKeyVal.(types.Tuple), t.sch)
+		if err != nil {
+			return errhand.BuildDError("failed to read table").AddCause(err).Build()
+		}
+		if rowExists {
+			return errors.New("cannot update primary key column")
```

Comment:
I would make this more specific, something like "Primary key collision" and give detail about which key is causing the issue.